### PR TITLE
wrangler e2e tests

### PR DESCRIPTION
**Pull request by AnkitCLI**
Wrangler e2e tests for excel ,xmltojson and json parsers.

### CODE REVIEW

File: wrangler-transform/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 1]
```diff
@@ -96,6 +101,51 @@ public static void deleteTempSourceBQTable() throws IOException, InterruptedExce
...
   }
+  @Before(order = 1, value = "@BQ_SOURCE_JSON_TEST")
+  public static void createTempSourceBQTableJson() throws IOException, InterruptedException {
+    createSourceBQTableWithQueries(PluginPropertyUtils.pluginProp("CreateBQTableQueryFileJson"),
+                                   PluginPropertyUtils.pluginProp("InsertBQDataQueryFileJson"));
+  }
+  @Before(order = 1, value = "@BQ_SOURCE_XML_TEST")
+  public static void createTempSourceBQTableXml() throws IOException, InterruptedException {
+    createSourceBQTableWithQueries(PluginPropertyUtils.pluginProp("CreateBQDataQueryFileXml"),
+            PluginPropertyUtils.pluginProp("InsertBQDataQueryFileXml"));
```
Comment (saimukkamala) [inline wrangler-transform/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: indentation is incorrect.

**Review by saimukkamala — APPROVED** [review]

File: wrangler-transform/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 1 above)
Comment (AnkitCLI) [reply wrangler-transform/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java]: done.

File: pom.xml [hunk 2]
```diff
@@ -83,7 +83,7 @@
     <aws.sdk.version>1.11.133</aws.sdk.version>
     <bigquery.connector.hadoop2.version>0.10.2-hadoop2</bigquery.connector.hadoop2.version>
     <bouncycastle.version>1.56</bouncycastle.version>
-    <cdap.version>6.10.0-SNAPSHOT</cdap.version>
+    <cdap.version>6.11.0-SNAPSHOT</cdap.version>
```
Comment (itsankit-google) [inline pom.xml]: why this change?

Comment (itsankit-google) [reply pom.xml]: Please revert if not required.

Comment (itsankit-google) [reply pom.xml]: Actually we can change it to 6.10.0 instead since it is released now.

**Review by itsankit-google — CHANGES_REQUESTED** [review]
Please use range of version in plugin versions and cdap data pipeline artifact version in the pipeline jsons otherwise we will need to always keep modifying the pipeline.

File: pom.xml (hunk 2 above)
Comment (AnkitCLI) [reply pom.xml]: changed to 6.10

**Comment by itsankit-google** [comment]
> Please use range of version in plugin versions and cdap data pipeline artifact version in the pipeline jsons otherwise we will need to always keep modifying the pipeline.

It does not looks like this comment is addressed, the plugin & cdap-data-pipeline versions are still hardcoded.

File: wrangler-transform/src/e2e-test/resources/testData/Wrangler/parse_csv_wrangle-cdap-data-pipeline.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
     "description": "Data Pipeline Application",
     "artifact": {
         "name": "cdap-data-pipeline",
-        "version": "6.10.0-SNAPSHOT",
+        "version": "[4.0.0, 5.0.0]",
```
Comment (itsankit-google) [inline wrangler-transform/src/e2e-test/resources/testData/Wrangler/parse_csv_wrangle-cdap-data-pipeline.json]: It should be [6.0.0, 7.0.0).

The comment applies to the whole PR.

Comment (AnkitCLI) [reply wrangler-transform/src/e2e-test/resources/testData/Wrangler/parse_csv_wrangle-cdap-data-pipeline.json]: done.

File: wrangler-transform/src/e2e-test/features/Wrangler/DataTypeParsers.feature [hunk 4]
```diff
@@ -24,8 +24,6 @@ Feature:  datatype parsers
     Then Replace input plugin property: "project" with value: "projectId"
     Then Replace input plugin property: "dataset" with value: "dataset"
     Then Replace input plugin property: "table" with value: "bqSourceTable"
-    Then Click on the Get Schema button
```
Comment (itsankit-google) [inline wrangler-transform/src/e2e-test/features/Wrangler/DataTypeParsers.feature:27]: why is this removed?

File: wrangler-transform/src/e2e-test/resources/testData/Wrangler/parse_HL7_Wrangler-cdap-data-pipeline (1).json [hunk 5]
```diff
@@ -3,7 +3,7 @@
     "description": "Data Pipeline Application",
     "artifact": {
         "name": "cdap-data-pipeline",
-        "version": "6.10.0-SNAPSHOT",
+        "version": "[6.0.0, 7.0.0]",
```
Comment (itsankit-google) [inline wrangler-transform/src/e2e-test/resources/testData/Wrangler/parse_HL7_Wrangler-cdap-data-pipeline (1).json]: nit: `[6.0.0, 7.0.0)`

this comment applies to whole PR.

Comment (AnkitCLI) [reply wrangler-transform/src/e2e-test/resources/testData/Wrangler/parse_HL7_Wrangler-cdap-data-pipeline (1).json]: changed in all the json pipelines.

File: wrangler-transform/src/e2e-test/features/Wrangler/DataTypeParsers.feature (hunk 4 above)
Comment (AnkitCLI) [reply wrangler-transform/src/e2e-test/features/Wrangler/DataTypeParsers.feature:27]: As the pipelines are exported.In almost all the PR's the issue is when we try to get schema then randomly in most of the fields it is going null when we click on get schema and validate button. And it also changes the schema . So just to solve that issue we removed that part from here.

Comment (itsankit-google) [reply wrangler-transform/src/e2e-test/features/Wrangler/DataTypeParsers.feature:27]: What do you mean by this?
```
when we try to get schema then randomly in most of the fields it is going null when we click on get schema and validate button. And it also changes the schema
```

Can you please elaborate with the help of screenshots?

File: wrangler-transform/src/e2e-test/resources/testData/Wrangler/parse_HL7_Wrangler-cdap-data-pipeline (1).json (hunk 5 above)
Comment (itsankit-google) [reply wrangler-transform/src/e2e-test/resources/testData/Wrangler/parse_HL7_Wrangler-cdap-data-pipeline (1).json]: This comment is not addressed still, the upper bound should not be included `[6.0.0, 7.0.0)`

**Review by itsankit-google — APPROVED** [review]

File: wrangler-transform/src/e2e-test/resources/testData/Wrangler/parse_HL7_Wrangler-cdap-data-pipeline (1).json (hunk 5 above)
Comment (AnkitCLI) [reply wrangler-transform/src/e2e-test/resources/testData/Wrangler/parse_HL7_Wrangler-cdap-data-pipeline (1).json]: changed.

File: pom.xml (hunk 2 above)
Comment (AnkitCLI) [reply pom.xml]: done

File: wrangler-transform/src/e2e-test/features/Wrangler/DataTypeParsers.feature (hunk 4 above)
Comment (AnkitCLI) [reply wrangler-transform/src/e2e-test/features/Wrangler/DataTypeParsers.feature:27]: added the schema and validation step.
